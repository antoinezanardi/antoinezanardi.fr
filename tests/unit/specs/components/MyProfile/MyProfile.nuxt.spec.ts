import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import { MyProfile } from "#components";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";

describe("My Profile Component", () => {
  let wrapper: ReturnType<typeof mount<typeof MyProfile>>;

  async function mountMyProfileComponent(options: ComponentMountingOptions<typeof MyProfile> = {}): Promise<ReturnType<typeof mount<typeof MyProfile>>> {
    return mountSuspendedComponent(MyProfile, { ...options });
  }

  beforeEach(async() => {
    wrapper = await mountMyProfileComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});