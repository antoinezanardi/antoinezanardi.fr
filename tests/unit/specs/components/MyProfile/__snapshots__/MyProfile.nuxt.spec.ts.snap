// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`My Profile Component > should match snapshot when rendered. 1`] = `
"<div data-v-d3c3f494="" class="profile-page">
  <div data-v-d3c3f494="" class="wrapper">
    <div data-v-d3c3f494="" class="page-header page-header-small" data-filter-color="green">
      <div data-v-d3c3f494="" class="page-header-image" data-parallax="true"></div>
      <div data-v-d3c3f494="" class="container">
        <div data-v-d3c3f494="" class="content-center">
          <div data-v-d3c3f494="" class="cc-profile-image"><a data-v-d3c3f494="" href="https://www.linkedin.com/in/antoinezanardi/" rel="noopener noreferrer" target="_blank">
              <nuxt-img-stub data-v-d3c3f494="" alt="Photo de profil" height="180" src="/images/antoine.webp" width="180" preload="false" crossorigin="false" placeholder="false" custom="false" id="profile-pic"></nuxt-img-stub>
            </a></div>
          <h1 data-v-d3c3f494="" class="h1 text-shadow title">Antoine Zanardi</h1>
          <h2 data-v-d3c3f494="" class="category mb-3 text-shadow text-white">Consultant Tech Lead</h2>
          <h2 data-v-d3c3f494="" class="category mb-3 text-shadow text-white">Expert Web FullStack</h2><a data-v-d3c3f494="" id="my-cv" class="border-2 border-success-subtle btn btn-primary rounded-3" data-aos="zoom-in" data-aos-once="true" download="CV Antoine ZANARDI.pdf" href="/antoine-zanardi-cv.pdf">Mon CV</a>
        </div>
      </div>
      <div data-v-d3c3f494="" class="section">
        <div data-v-d3c3f494="" class="container">
          <div data-v-d3c3f494="" class="button-container" data-aos="fade-up" data-aos-anchor-placement="top-bottom" data-aos-once="true">
            <linked-in-button-stub data-v-d3c3f494="" class="me-2"></linked-in-button-stub>
            <git-hub-button-stub data-v-d3c3f494=""></git-hub-button-stub>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
